.github/workflows/build-package.yml.template

```diff
@@ -110,15 +110,16 @@ jobs:
         run: |
           (
             cd trigger
-            git tag $CRATE-$VERSION-$TARGET_ARCH
-            git push origin $CRATE-$VERSION-$TARGET_ARCH
+            git tag $CRATE-$VERSION
+            git push origin $CRATE-$VERSION
```

Comment:
either of these commands might fail now (depending on whether our checkout action also pulls tags+history -  I can't remember)

If they do then just print a message saying this is fine and carry on.

Reply:
Not sure about why this would fail though, since git tag should succeed unless the tag already exists

Reply:
building bat 0.5 for linux will create `bat-0.5` as a tag

build bat 0.5 for macos will try to create `bat-0.5` as a tag but it already exists (or if it doesn't exist locally on the runner due to a sparse checkout, pushing a tag that already exists should fail)

This wasn't a problem with the old code.

Reply:
> building bat 0.5 for linux will create `bat-0.5` as a tag
> 
> build bat 0.5 for macos will try to create `bat-0.5` as a tag but it already exists (or if it doesn't exist locally on the runner due to a sparse checkout, pushing a tag that already exists should fail)
> 
> This wasn't a problem with the old code.

I was aware of this, which is why I switch to a different release action, the same one used in cargo-binstall where we upload multiple artifacts from different workflow runs to the same tag.

Reply:
this block of code will fail purely because of the rules of git. This fine, but you need to catch the error

...I've just noticed that we don't have a `set -e` in this block, so it might just carry on if the first command fails (`git tag`, and only bomb out if the second one fails (`git push`). In that case, it may be that this will 'just work". I guess we'll just have to try it and see.

Reply:
Perhaps I should add a set -euxo, it's better to fail ASAP on the first cmd failure

Reply:
I've added `set -euxo`.

Reply:
Hmm I forgot that `set -euxo` doesn't work on windows.

Reply:
Should work if you put it as the first line of your script block, I think. I've not looked into your build errors yet. Will do in an hour or so when I'm properly awake.

Reply:
> Should work if you put it as the first line of your script block, I think. I've not looked into your build errors yet. Will do in an hour or so when I'm properly awake.

I've tried [this](https://github.com/cargo-bins/cargo-quickinstall/actions/runs/4333603752/jobs/7566866862) but it fails on windows.

Reply:
You probably want `shell: bash` to make sure windows doesn't use powershell, and then `set -euxo pipefail` inside the script.

Reply:
@alsuren Thanks, that works!